### PR TITLE
Testing get total supply

### PR DESCRIPTION
**Pull request by BenMacnaughton**

### CODE REVIEW

File: src/createCalls/createCalls.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+const ajv = new Ajv();
+
+interface CallParams {
+  mintPrice: number;
+  mintAmount?: number;
+  contractAddress: string;
+  minterAddress: string;
+  jsonRpcUrl?: string;
+  networkId?: number;
```
Comment (kasahenry) [inline src/createCalls/createCalls.ts:14]: Not sure what this is, is this `chainId` ?

File: src/createCalls/createCalls.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    networkId: { type: 'number', minimum: 1, maximum: 100000 },
+    userAddress: {
+      type: 'string',
+      pattern: '^0x[a-fA-F0-9]{40}$',
+      default: '0x0123456789abcdeffedcba9876543210deadbeef',
+    },
+  },
+  required: ['mintPrice', 'mintAmount', 'contractAddress', 'minterAddress'],
+  anyOf: [{ required: ['jsonRpcUrl'] }, { required: ['networkId'] }],
+};
```
Comment (kasahenry) [inline src/createCalls/createCalls.ts:35]: this is kind of cool, what happens if you fail these checks ?

File: src/createCalls/createCalls.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+export type ICall = {
+  _to: string;
+  _value: any;
+  _calldata: string;
+};
+
+// Regular Mint ICall
+export function CreateMintICall(callParams: CallParams): ICall[] {
```
Comment (kasahenry) [inline src/createCalls/createCalls.ts:44]: super nitpick but probably camelCase here

**Review by kasahenry — COMMENTED** [review]
Few very minor comments, looks good

File: src/createCalls/createCalls.ts (hunk 1 above)
Comment (BenMacnaughton) [reply src/createCalls/createCalls.ts:14]: Fixed

File: src/createCalls/createCalls.ts (hunk 2 above)
Comment (BenMacnaughton) [reply src/createCalls/createCalls.ts:35]: We can customize the error message, but it just rejects the params and throws a general error atm

File: src/createCalls/createCalls.ts (hunk 3 above)
Comment (BenMacnaughton) [reply src/createCalls/createCalls.ts:44]: Fixed

File: src/createCalls/minterABIs.ts [hunk 4]
```diff
@@ -0,0 +1,370 @@
+import { ContractInterface } from 'ethers';
+
+export const minterABIs: { [address: string]: ContractInterface } = {
+  '0x204a879975378ac0b1f8f378f8cceadc75422f97': [
```
Comment (greguezono) [inline src/createCalls/minterABIs.ts:4]: Every time we have a new integration, would we have to add it into this minterABIs map?
That's fine, just for my own context

File: src/createCalls/contractABIs.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { ContractInterface } from 'ethers';
+
+export const contractABIs: { [address: string]: ContractInterface } = {};
```
Comment (greguezono) [inline src/createCalls/contractABIs.ts:3]: Is this map for our delegate contracts?

File: src/createCalls/createCalls.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      type: 'string',
+      pattern: '^0x[a-fA-F0-9]{40}$',
+      default: '0x0123456789abcdeffedcba9876543210deadbeef',
+    },
+  },
+  required: ['mintPrice', 'mintAmount', 'contractAddress', 'minterAddress'],
+  anyOf: [{ required: ['jsonRpcUrl'] }, { required: ['chainId'] }],
+};
+
+export type ICall = {
```
Comment (greguezono) [inline src/createCalls/createCalls.ts:37]: Can you import this from our widget or direct-nft-protocol repo instead of redeclaring?

Comment (greguezono) [reply src/createCalls/createCalls.ts:37]: Preferably direct-nft-protocol. I think it's exported from there but not sure.

**Review by greguezono — APPROVED** [review]
Good stuff, just some general questions and a small thing to import

File: src/createCalls/minterABIs.ts (hunk 4 above)
Comment (BenMacnaughton) [reply src/createCalls/minterABIs.ts:4]: Yep exactly

File: src/createCalls/contractABIs.ts (hunk 5 above)
Comment (greguezono) [reply src/createCalls/contractABIs.ts]: It's all good, contract is a more general term so keep it as is

**Pull request merged**